creator-node/src/serviceRegistry.js

```diff
@@ -0,0 +1,77 @@
+const redisClient = require('./redis')
+const { ipfs, ipfsLatest } = require('./ipfsClient')
+const BlacklistManager = require('./blacklistManager')
+const AudiusLibs = require('@audius/libs')
+const config = require('./config')
+
+const initAudiusLibs = async () => {
+  const ethWeb3 = await AudiusLibs.Utils.configureWeb3(
+    config.get('ethProviderUrl'),
+    config.get('ethNetworkId'),
+    /* requiresAccount */ false
+  )
+  const dataWeb3 = await AudiusLibs.Utils.configureWeb3(
+    config.get('dataProviderUrl'),
+    null,
+    false
+  )
+  const discoveryProviderWhitelist = config.get('discoveryProviderWhitelist')
+    ? new Set(config.get('discoveryProviderWhitelist').split(','))
+    : null
+  const identityService = config.get('identityService')
+
+  const audiusLibs = new AudiusLibs({
+    ethWeb3Config: AudiusLibs.configEthWeb3(
+      config.get('ethTokenAddress'),
+      config.get('ethRegistryAddress'),
+      ethWeb3,
+      config.get('ethOwnerWallet')
+    ),
+    web3Config: {
+      registryAddress: config.get('dataRegistryAddress'),
+      useExternalWeb3: true,
+      externalWeb3Config: {
+        web3: dataWeb3,
+        ownerWallet: config.get('delegateOwnerWallet')
+      }
+    },
+    discoveryProviderConfig: AudiusLibs.configDiscoveryProvider(discoveryProviderWhitelist),
+    // If an identity service config is present, set up libs with the connection, otherwise do nothing
+    identityServiceConfig: identityService ? AudiusLibs.configIdentityService(identityService) : undefined,
+    isDebug: config.get('creatorNodeIsDebug')
+  })
+  await audiusLibs.init()
+  return audiusLibs
+}
+
+/**
+ * Construct a ServiceRegistry object.
+ *
+ * `initServices` must be called before services are consumed.
+ */
+function ServiceRegistry () {
```

Comment:
should this be a class?

Reply:
also 'Service' is a bit overloaded, would it be wrong to call this a ComponentRegistry (or something else)?

Reply:
+1

Reply:
Happy to make it a class for consistency; I prefer constructor functions (and it feels a bit weird to deconstruct a class in the controller argument, but it works). 

Re: naming, I don't think `ComponentRegistry` is quite what we want - a `Component` in this architecture is sort of like an app 'feature' - a bundle of routes, business logic, tests, etc. Having a 'Service Layer' is a pretty standard Node architecture (https://blog.logrocket.com/the-perfect-architecture-flow-for-your-next-node-js-project/). I'm down to call it something else but imo it shouldn't be component, and I think Service is the best fit, also matching what I've seen on previous teams/other projects.

Reply:
eh, i'm good with function as long as it works. just not used to seeing it. i think i just had same question - if making this a function instead of class was intentional

Reply:
Made it a class for consistency